### DISCUSSION TITLE
How do I add multiple translation files?

### DISCUSSION THREAD

**Discussion by misaka10843** [WeblateOrg/weblate#9723] [Q&A] [answered]
The list of files in my native language looks like this

```
docs/en
│  faq.md
│  index.md
│
├─helping-the-server
│  │  for-developers.md
│  │
│  ├─Art
│  │      aboutartteam.md
│  │      artistguides.md
│  │
│  ├─Building
│  │      about-the-build-team.md
│  │      guides.md
│  │
│  └─Design
│          about-the-design-team.md
│
├─rules
│      bancircumvention.md
│      disrespectingothers.md
│      excessivepvp.md
│      exploitabuse.md
│      grief.md
│      hacking.md
│      moderationdecisions.md
│      rules.md
│      spam.md
│      Swear&Hate.md
│
└─server-guide
        bosses.md
        classes-skills.md
        crop-resources.md
        drop-resources.md
        fishing-resources.md
        money.md
        ore-rates.md
        starting-out.md
        touhou-mobs.md
        wildlife-mobs.md
```
Then I use Component discovery to add components automatically

![image](https://github.com/WeblateOrg/weblate/assets/69132853/30f4b3f8-fff4-46d9-a9f0-930d2c2d3478)
![image](https://github.com/WeblateOrg/weblate/assets/69132853/1910c05e-84b6-4e5c-9fb8-8838b02a2aee)

But after saving, clicking on any Component discovery generates a component with no strings

![image](https://github.com/WeblateOrg/weblate/assets/69132853/e31faa66-86d3-4104-ab67-9f19673787c9)

How can I fix it?

Another question is how do I add new languages to all the components?

---

weblate version: Docker [bleeding-2023-08-10-3bf021a3bb1be8a4e51d33db4bd263ad14c2508d](https://hub.docker.com/layers/weblate/weblate/bleeding-2023-08-10-3bf021a3bb1be8a4e51d33db4bd263ad14c2508d/images/sha256-bae04c43fb9c86d5fe8c19fc68fa494062e2a5fbc844b2b6595c59b9e7b5e155?context=explore)

**Comment by nijel** [comment, marked as answer]
You need a monolingual base file here.

**Reply by misaka10843** [reply]
Thank you, because I completed a monolingual base file similar to·(?P<originalHierarchy>.+/)· results in an error, so this item can be null 🤣

**Reply by nijel** [reply]
It is required here, I've added missing validation in 0cade25731d04e34cfa495290f9396037b0514f8.

You will need something like `docs/en/{{ originalHierarchy }}/{{ component }}.md`
